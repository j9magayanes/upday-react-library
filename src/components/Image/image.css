

img {

        min-width: 300px;
        max-width: 300px;
        min-height: 250px;
        max-height: 250px;
    
}