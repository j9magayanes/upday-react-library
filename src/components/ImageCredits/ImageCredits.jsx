import React from 'react';
import './imageCredits.css'

export const ImageCredits = ({ ...props}) => {
    return (
      <span className="imageCredits">Baden-Württemberg, Ulm: Der Angeklagte wird in Hand- und Fußschellen in den Gerichtssaal geführt. Foto: Felix Kästle/dpa</span>
    )
}