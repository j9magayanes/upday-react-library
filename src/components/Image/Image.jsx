import React from 'react';
import './image.css';
import 'upday-web-components/components/imageComponent'

export const Image = ({label, src, ...args}) => {
    return (


        <figure>
          <image-component src={src} label={ args.title } size={args.size} class={"meinestadt"}/>
      </figure>
    ) 
}