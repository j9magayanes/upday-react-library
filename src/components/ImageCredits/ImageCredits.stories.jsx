import React from 'react';

import { ImageCredits } from './ImageCredits';


export default {
  title: 'Form/ImageCredits',
  component:ImageCredits,
};

const Template = () =><ImageCredits  />;


export const Default = Template.bind({});