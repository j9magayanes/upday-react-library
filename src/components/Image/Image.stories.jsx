import React from 'react';

import { Image } from './Image';


export default {
  title: 'Form/Image',
  component: Image,
};

const Template = (args) =><Image {...args} />;


export const Default = Template.bind({});

export const Meinestadt = Template.bind({});
Meinestadt.args = {
  label: 'Meinestadt',
  title: 'Mehr als 16 Monate nach Beginn des russischen Angriffskriegs hat der ukrainische Präsident Wolodymyr Selenskyj die Forderung nach einer baldigen konkreten EU-Beitrittsperspektive für sein Land erneuert. „Die Ukraine hat Einfluss auf die Stärke Europas. Das ist ein Fakt“, sagte Selenskyj am Donnerstag in seiner abendlichen Videoansprache. „Und dieses Jahr ist es an der Zeit, diesen und andere Fakten zu nutzen, um die Einheit in Europa zu stärken – angefangen beim Start von Verhandlungen über die EU-Mitgliedschaft der Ukraine.“ Unterdessen reiste der frühere US-Vizepräsident Mike Pence in die Ukraine. Von Selenskyj in Kiew empfangen wurde zudem die schwedische Umweltaktivistin Greta Thunberg.',
  size: 'meinestadt',
  args:{   src: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcTnFYE7C4Uz5vuWSQUmIsny2oz4A9WFbSrJaQ&usqp=CAU',}
};

export const Web = Template.bind({});
Web.args = {
  title: 'Mehr als 16 Monate nach Beginn des russischen Angriffskriegs hat der ukrainische Präsident Wolodymyr Selenskyj die Forderung nach einer baldigen konkreten EU-Beitrittsperspektive für sein Land erneuert. „Die Ukraine hat Einfluss auf die Stärke Europas. Das ist ein Fakt“, sagte Selenskyj am Donnerstag in seiner abendlichen Videoansprache. „Und dieses Jahr ist es an der Zeit, diesen und andere Fakten zu nutzen, um die Einheit in Europa zu stärken – angefangen beim Start von Verhandlungen über die EU-Mitgliedschaft der Ukraine.“ Unterdessen reiste der frühere US-Vizepräsident Mike Pence in die Ukraine. Von Selenskyj in Kiew empfangen wurde zudem die schwedische Umweltaktivistin Greta Thunberg.',
  label: 'Web',
  size: 'web',
  args:{   src: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcTnFYE7C4Uz5vuWSQUmIsny2oz4A9WFbSrJaQ&usqp=CAU',}
};