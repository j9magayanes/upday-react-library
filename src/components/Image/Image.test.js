// ImageComponent.test.js
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import ImageCredits  from 'upday-web-components/components/imageCredits';
import ImageComponent from 'upday-web-components/components/imageComponent';
import { Image } from './Image';
import { JSDOM } from 'jsdom';

// Import your custom web component class
// Assuming the web component file is in the same directory

describe('ImageComponent', () => {
  test('renders with default attributes', () => {
    // Create a virtual DOM
    const dom = new JSDOM('<!DOCTYPE html>');
    global.document = dom.window.document;

    // Instantiate your web component
    const imageComponent = <ImageComponent />;


    // Perform assertions based on the default state
    expect(imageComponent.shadowRoot).toBeDefined();
    //expect(imageComponent.imgElement).toBeDefined();
    //expect(imageComponent.imgElement.getAttribute('src')).toBeNull(); // No 'src' attribute initially
    // Add more assertions as needed for other default attributes
  });

  test('updates attributes', () => {
    // Create a virtual DOM
    const dom = new JSDOM('<!DOCTYPE html>');
    global.document = dom.window.document;

    // Instantiate your web component
    const imageComponent = document.createElement('image-component');
    document.body.appendChild(imageComponent);

    // Update attributes
    imageComponent.setAttribute('src', 'example.jpg');
    imageComponent.setAttribute('title', 'Example Title');
    imageComponent.setAttribute('label', 'Example Label');
    imageComponent.setAttribute('size', 'large');
    imageComponent.setAttribute('class-prop', 'custom-class');

    // Perform assertions after attribute updates
    //expect(imageComponent.imgElement.getAttribute('src')).toBe('example.jpg');
    //expect(imageComponent.imgElement.getAttribute('title')).toBe('Example Title');
    //expect(imageComponent.imgElement.getAttribute('label')).toBe('Example Label');
    //expect(imageComponent.imgElement.getAttribute('size')).toBe('large');
    //expect(imageComponent.contentElement.className).toBe('custom-class');
  });

  // Add more test cases as needed
});

test('ImageComponent renders correctly', () => {
  const props = {
    label: 'Example Label',
    src: 'example.jpg',
    title: 'Example Title',
    size: 'large',
  };

  const component = renderer.create(<Image {...props} />);
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});