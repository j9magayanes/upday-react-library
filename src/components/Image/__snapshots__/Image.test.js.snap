// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageComponent renders correctly 1`] = `
<figure>
  <image-component
    class="meinestadt"
    label="Example Title"
    size="large"
    src="example.jpg"
  />
</figure>
`;
